localstack-core/localstack/services/secretsmanager/provider.py

```diff
@@ -173,9 +173,20 @@ def create_secret(
         self, context: RequestContext, request: CreateSecretRequest
     ) -> CreateSecretResponse:
         self._raise_if_missing_client_req_token(request)
-        self._raise_if_invalid_secret_id(request["Name"])
+        # Some providers need to create keys which are not usually creatable by users
+        if not any(
+            tag_entry["Key"] == "BYPASS_SECRET_ID_VALIDATION"
+            for tag_entry in request.get("Tags", [])
+        ):
+            self._raise_if_invalid_secret_id(request["Name"])
+        else:
+            request["Tags"] = [
+                tag_entry
+                for tag_entry in request.get("Tags", [])
+                if tag_entry["Key"] != "BYPASS_SECRET_ID_VALIDATION"
+            ]
```

Comment:
We sadly need something to bypass the name validation here, as eventbridge connection secrets contain a `!`, which is not allowed. I used a tag here, but there are other options should we decide against it.

Reply:
>  which is not allowed

I'm not sure I'm following here. Not allowed as in our SecretsManager service *should* support it, but doesn't? Or how would this otherwise work on AWS?

Reply:
Are AWS services able to create keys with names that we (as external users) can't?

Reply:
It seems that way. The error message clearly states it should not be allowed, the aws console also does not allow it. So maybe aws services have some bypass for it.